### DISCUSSION TITLE
Require instructor approval for feedback pull request merge

### DISCUSSION THREAD

**Discussion by AlecMcT** [community/community#147258] [GitHub Education]
### Hi everyone,

Students can currently complete the feedback pull request for a GitHub Classroom assignment, thus cutting off the crucial feedback/review avenue between teacher and student, with no way to reopen it / reverse the merge. I was hoping to find a way to add a rule - at the assignment creation/distribution level to "Require approval from specific reviewers" (i.e. the instructor) "before merging" - as shown in the tip above each student's merge button for the feedback pull request. I assume I can manually add that rule to individual repositories; is it possible to have that rule present from the start, based on assignment / template repository settings?

Since this happened with one of my 24 high school students on the 2nd major assignment, I'm assuming that premature closing of the Feedback pull request must happen on occasion for most classes/instructors. This single rule/feature should block even the possibility of that happening.

**Comment by Genyus** [comment]
Late reply, but as I recently resolved this same issue for my team, I'm sharing my findings in case it's of use to anyone else encountering this issue.

To apply some configuration across every student repo automatically, you'll need a [GitHub App](https://docs.github.com/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps) to be added to your Classroom organisation.

There's a [Probot](https://probot.github.io/docs/) [community app](https://probot.github.io/apps/mergeable/) that enables configuration-based merge blocking, but as we have some more specific requirements, I built our own. This should meet the immediate need shared above, however.
